### ISSUE TITLE
Password Protected Feeds

### ISSUE THREAD

**Issue by satisficer** [manolomartinez/greg#49]
I have a password protected feed that I can't seem to get to work. I've tried putting `user = xxxx` and `password = yyyy` in `greg.conf` and also passing the username and password into the url, i.e. `https://user:password@www.dev.invalid/` (there is a trailing slash in feed url). Neither approach has worked so far. Is there a simple approach that I've missed? I can provide more specific information if needed.

**Comment by manolomartinez** [comment]
Huh, this is a new one. Greg currently has no capabilities to incorporate login information, mostly because this is the first time a user happens upon a password-protected feed. So, for the time being the easiest thing would be to add user and password to the url, as you were trying to do. How do you normally access this feed? Through a web interface, or another program?

**Comment by satisficer** [comment]
On Fri, May 13, 2016 at 12:50:14AM -0700, manolomartinez wrote:

> So, for the time being the easiest thing would be to add user and password to the url, as you were trying to do.

I will keep trying to get this to work and let you know if it does :)

> How do you normally access this feed? Through a web interface, or another program?

It's the first feed I've ever had that needs authentication. I tried the methods in the [`feedparser` documentation](https://pythonhosted.org/feedparser/http-authentication.html) to see if I could get it to work. Clementine did not handle the login, but I did get it to work on gpodder3 in Arch and in AntennaPod and PodcastAddict on Android. All the working methods have a dialog that pops up to enter `user`/`password` information.

Thanks for `greg` -- it's the best podcast client around!

**Comment by ChaosTherum** [comment]
Yeah this would be quite useful I have 2 different premium subscriptions that I would like to use with greg.

**Comment by usmanatron** [comment]
I know this is quite old but I'm really keen to get password-protected feeds definitely working (only thing stopping me from using greg).

I'd like to have a go at implementing something - @manolomartinez do you have any initial thoughts \ preferences on implementation?  In the meantime I'll have a play :-)

**Comment by manolomartinez** [comment]
Hey, @usmanatron, thanks for your interest in helping me with greg :)

Can you retrieve your password-protected feed, e.g., by using `wget --user=user --password=password url`? If so, it should be as easy as using a custom `downloadhandler` in `greg.config`.
